src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,497 @@
+
+    -- * Relational operations
+    compose, -- closure, reflexiveClosure, symmetricClosure, transitiveClosure
   ) where
 
-import Algebra.Graph.Relation
+import qualified Algebra.Graph.Relation as R
+import qualified Algebra.Graph.Relation.Internal as RI hiding (empty, vertex, overlay, connect, referredToVertexSet)
 import Algebra.Graph.Relation.InternalDerived
 
 import qualified Data.Set as Set
+import Data.Tuple
+import Data.Tree 
+import qualified Data.Tree as Tree
 
 -- | Construct a symmetric relation from a 'Relation'.
 -- Complexity: /O(1)/ time.
-fromRelation :: Relation a -> SymmetricRelation a
+fromRelation :: R.Relation a -> SymmetricRelation a
 fromRelation = SymmetricRelation
 
 -- | Extract the underlying relation.
 -- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
-toRelation = symmetricClosure . fromSymmetric
+toRelation :: Ord a => SymmetricRelation a -> R.Relation a
+toRelation = R.symmetricClosure . fromSymmetric
+
+-- | Construct the graph comprising /a single edge/.
+-- Complexity: /O(1)/ time, memory and size.
+--
+-- @
+-- edge x y                 == edges [(1,2),(2,1)]
+-- y)) ('connect' ('vertex' y) ('vertex' x))
+-- 'hasEdge' x y (edge x y) == True
+-- 'edgeCount'   (edge x y) <= 2
+-- 'vertexCount' (edge 1 1) == 1
+-- 'vertexCount' (edge 1 2) == 2
+-- @
+edge :: Ord a => a -> a -> SymmetricRelation a
+edge x y = SymmetricRelation $ RI.Relation (Set.fromList [x, y]) (Set.fromList [(x, y), (y,x)]) 
+
+-- | Construct the graph comprising a given list of isolated vertices.
+-- Complexity: /O(L * log(L))/ time and /O(L)/ memory, where /L/ is the length
+-- of the given list.
+--
+-- @
+-- vertices []            == 'empty'
+-- vertices [x]           == 'vertex' x
+-- 'hasVertex' x . vertices == 'elem' x
+-- 'vertexCount' . vertices == 'length' . 'Data.List.nub'
+-- 'vertexSet'   . vertices == Set.'Set.fromList'
+-- @
+vertices :: Ord a => [a] -> SymmetricRelation a
+vertices = SymmetricRelation . R.vertices
+
+-- | Construct the graph from a list of edges.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- edges []          == 'empty'
+-- edges [(x,y)]     == 'overlay' ('edge' x y) ('edge' y x)
+-- 'edgeCount' . edges == 'length' . 'Data.List.nub' . 'Data.List.union'
+-- . '<map swap, id>'
+-- @
+edges :: Ord a => [(a, a)] -> SymmetricRelation a
+edges es = SymmetricRelation $ RI.Relation (Set.fromList $ uncurry (++) $ unzip es) (Set.fromList es `Set.union` Set.map swap (Set.fromList es))
+
+-- | Overlay a given list of graphs.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- overlays []        == 'empty'
+-- overlays [x]       == x
+-- overlays [x,y]     == 'overlay' x y
+-- overlays           == 'foldr' 'overlay' 'empty'
+-- 'isEmpty' . overlays == 'all' 'isEmpty'
+-- @
+overlays :: Ord a => [SymmetricRelation a] -> SymmetricRelation a
+overlays = SymmetricRelation . R.overlays . map fromSymmetric
```

Comment:
As mentioned in a another comment, all these `fromSymmetric` will be gone if you introduce an additional invariant for `SymmetricRelation` that it stores edges in both directions.

Reply:
Already answered